flow/src/main/java/flow/Flow.java

```diff
@@ -16,27 +16,45 @@
 
 package flow;
 
+import android.app.Activity;
 import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.Nullable;
 import android.view.View;
 import java.util.Iterator;
 
+import static flow.Preconditions.checkArgument;
 import static flow.Preconditions.checkNotNull;
 
 /** Holds the current truth, the history of screens, and exposes operations to change it. */
 public final class Flow {
-  private static final String FLOW_SERVICE = "flow.Flow.FLOW_SERVICE";
+  static final String HISTORY_KEY = InternalLifecycleIntegration.class.getSimpleName() + "_history";
 
   public static Flow get(View view) {
     return get(view.getContext());
   }
 
   public static Flow get(Context context) {
-    //noinspection ResourceType
-    return (Flow) context.getSystemService(FLOW_SERVICE);
+    return InternalContextWrapper.getFlow(context);
   }
 
-  public static boolean isFlowSystemService(String name) {
-    return FLOW_SERVICE.equals(name);
+  public static Installer configure(Context baseContext, Activity activity) {
+    return new Installer(baseContext, activity);
+  }
+
+  public static boolean onBackPressed(Activity activity) {
```

Comment:
What do you mean?  How many apps won't want to respond to the back button by invoking Flow?

It _is_ prescriptive, and its name intentionally implies "call this from `onBackPressed`.If you're doing clever things with back button handling, you're clever enough to decide when/whether to actually make this call. If you're just getting started, you should definitely call it until/unless you know otherwise.